src/game/default/g_client_stats.c

```diff
@@ -270,6 +270,12 @@ void G_ClientStats(g_entity_t *ent) {
 		client->ps.stats[STAT_READY] = client->locals.persistent.ready;
 	}
 
+	// respawn time
+	client->ps.stats[STAT_RESPAWN] = 0;
+	if (g_level.time < client->locals.respawn_time) {
+		client->ps.stats[STAT_RESPAWN] = client->locals.respawn_time - g_level.time;
```

Comment:
Off the top of my head, I don't think the client knows about server time.
The timer and match state (timeout, warmup, countdown) is sent as a config string to all clients.